llvm/lib/TargetParser/Triple.cpp

```diff
@@ -1151,6 +1151,13 @@ std::string Triple::normalize(StringRef Str) {
     }
   }
 
+  // For 3-component triples, the middle component is used to set Vendor;
```

Comment:
Suggested comment

If "none" is in the middle component in a three-component triple, treat it as
the OS (Components[2]) instead of the vendor (Components[1]).